projects/cdk/services/theme-color.service.ts

```diff
@@ -0,0 +1,49 @@
+import {DOCUMENT} from '@angular/common';
+import {inject, Injectable} from '@angular/core';
+import {tuiCreateTokenFromFactory} from '@taiga-ui/cdk/utils';
+
+export const TUI_THEME_COLOR = tuiCreateTokenFromFactory(() =>
+    getColor(inject(DOCUMENT)),
```

Comment:
I don't think we should get it from meta or CSS variable, we should instead default it too nothing and allow people to provide it through token, which would in turn set it in meta and in CSS variable.

Reply:
updated